src/Nethermind/Nethermind.Core/BlockInfo.cs

```diff
@@ -19,6 +19,13 @@
 
 namespace Nethermind.Core
 {
+    public enum FinalizationStatus
+    {
+        None = 0,
+        Finalized = 1,
+        Invalid,
+    }
```

Comment:
1. Should be [Flags], we want to use it more than just finalization.
2. Rename to BlockMetadata, its will only about finalization.
3. Use explicit values in the enum

Reply:
Plugin-extensible could be messy. What if we switch from one plugin to another with different set of flags?

Reply:
Yes there are potential conflicts on only 32 or 64 flags